worlds/psychonauts/Regions.py

```diff
@@ -0,0 +1,838 @@
+        LocationName.NearDiegosHouseMaxLives,
+    ],
+    RegionName.BVRBLev: [
+        LocationName.ClubStreetMetalBalconyDufflebagTag,
+        LocationName.HeartStreetHIGHBalconyAmmoUp,
+    ],
+    RegionName.BVRBTele: [
+        LocationName.ClubStreetGatedSteamerTrunkTag,
+        LocationName.TheGardenVault,
+    ],
+    RegionName.BVRBDuster: [
+        LocationName.DiegosBedSuitcaseTag,
+        LocationName.DiegosRoomHatbox,
+        LocationName.DiegosHouseGrindrailSuitcase,
+        LocationName.GrindrailBalconyConfusionAmmoUp,
+    ],
+    RegionName.BVRBLogs: [
+        LocationName.BurnTheLogsDufflebag,
+    ],
+    RegionName.BVES: [
+        LocationName.SanctuaryGroundPurse,
+        LocationName.TigerWrestler,
+        LocationName.DragonWrestler,
+        LocationName.EagleWrestler,
+
+    ],
+    RegionName.BVESLev: [
+        LocationName.SanctuaryBalconyPurseTag,
+    ],
+    RegionName.BVESCobra: [
+        LocationName.CobraWrestler,
+    ],
+    RegionName.BVESBoss: [
+        LocationName.BlackVelvetopiaComplete,
+    ],
+
+    # Meat Circus
+    RegionName.MCTC: [
+        LocationName.CrumblingPathSteamertrunk,
+        LocationName.CrumblingPathEndRightHatboxTag,
+        LocationName.CrumblingPathEndLeftConfusionAmmoUp,
+        LocationName.OllieEscortFloorSuitcaseTag,
+    ],
+    RegionName.MCTCLev: [
+        LocationName.EntranceAwningSteamertrunkTag,
+    ],
+    RegionName.MCTCEscort: [
+        LocationName.OllieEscortMiddleHatbox,
+        LocationName.OllieEscortTopLeftVault,
+        LocationName.OllieEscortTopRightPurseTag,
+        LocationName.TunnelOfLoveStartPurse,
+        LocationName.TunnelOfLoveCornerSuitcase,
+        LocationName.TunnelOfLoveRailDufflebagTag,
+        LocationName.NextToTheFatLadyDufflebag,
+    ],
+}
+
+
+def place_events(self: "PSYWorld"):
+    final_boss_location = self.multiworld.get_location(LocationName.MeatCircusFinalBossEvent, self.player)
+    oleander_boss_location = self.multiworld.get_location(LocationName.OleanderBrainTankBossEvent, self.player)
+    redeemed_required_brains = self.multiworld.get_location(LocationName.RedeemedBrainsEvent, self.player)
+    # Meat Circus Bosses
+    if self.options.RequireMeatCircus:
+        victory = final_boss_location
+    else:
+        # Brain Hunt
+        if self.options.Goal == Options.Goal.option_brain_hunt:
+            victory = redeemed_required_brains
+        # Brain Tank or Brain Tank AND Brain Hunt
+        # For the Brain Tank AND Brain Hunt goal, both must be completed for victory. While the player can complete both
+        # goals in any order, logically, the Brain Tank goal can only be completed after the Brain Hunt goal.
+        else:
+            victory = oleander_boss_location
+
+    for location in (final_boss_location, oleander_boss_location, redeemed_required_brains):
+        if location is victory:
+            event_name = ItemName.Victory
+        else:
+            event_name = ItemName.Filler
```

Comment:
The locations exist because the enabled goals must logically be completed in sequence, with the last goal in the sequence resulting in victory. Removing the events for disabled goals probably could be done, but would require changes elsewhere.